File: python/semantic_kernel/filters/filter_context_base.py
# Copyright (c) Microsoft. All rights reserved.

from typing import TYPE_CHECKING

from semantic_kernel.kernel_pydantic import KernelBaseModel

if TYPE_CHECKING:
    from semantic_kernel.functions.kernel_arguments import KernelArguments
    from semantic_kernel.functions.kernel_function import KernelFunction
    from semantic_kernel.kernel import Kernel


class FilterContextBase(KernelBaseModel):
    """Base class for Kernel Filter Contexts."""

    function: "KernelFunction"
    kernel: "Kernel"
    arguments: "KernelArguments"
    is_streaming: bool = False

File: nodejs/semantic_kernel/filters/filter-context-base.ts
import { Kernel, KernelFunction } from '../kernel'
import { KernelArguments } from '../functions/kernel-arguments'

/**
 * Base class for Kernel Filter Contexts.
 */
export abstract class FilterContextBase {
  /** The function being invoked */
  function: KernelFunction

  /** The kernel instance */
  kernel: Kernel

  /** The arguments passed to the function */
  arguments: KernelArguments

  /** Whether the function is streaming */
  isStreaming: boolean

  constructor(options: {
    function: KernelFunction
    kernel: Kernel
    arguments: KernelArguments
    isStreaming?: boolean
  }) {
    this.function = options.function
    this.kernel = options.kernel
    this.arguments = options.arguments
    this.isStreaming = options.isStreaming ?? false
  }
}
